softwareComponents/configuration/json_include/configuration/serialization.hpp

```diff
@@ -448,46 +450,50 @@ namespace rofi::configuration::serialization {
         // function for translation of components to docs if necessary
         auto f = [ &world ]( ModuleId id ) -> ModuleType { return world.getModule( id )->type; };
 
-        for ( const auto& jj : j[ "moduleJoints" ] ) {
-            roficom::Orientation o = roficom::stringToOrientation( jj[ "orientation" ] )
-                                             .get_or_throw_as< std::runtime_error >();
+        if ( j.contains( "moduleJoints" ) ) {
+            for ( const auto& jj : j[ "moduleJoints" ] ) {
+                roficom::Orientation o = roficom::stringToOrientation( jj[ "orientation" ] )
+                                                .get_or_throw_as< std::runtime_error >();
 
-            auto [ sourceModule, sourceConnector ] = details::connectionComponentFromJSON( jj[ "from" ], f );
-            auto [ destinationModule, destinationConnector ] = details::connectionComponentFromJSON( jj[ "to" ], f );
+                auto [ sourceModule, sourceConnector ] = details::connectionComponentFromJSON( jj[ "from" ], f );
+                auto [ destinationModule, destinationConnector ] = details::connectionComponentFromJSON( jj[ "to" ], f );
 
-            auto conn = connect( world.getModule( sourceModule )->connectors()[ sourceConnector ]
-                               , world.getModule( destinationModule )->connectors()[ destinationConnector ]
-                               , o );
+                auto conn = connect( world.getModule( sourceModule )->connectors()[ sourceConnector ]
+                                , world.getModule( destinationModule )->connectors()[ destinationConnector ]
+                                , o );
 
-            details::processAttributes( jj, attrCb, conn );
+                details::processAttributes( jj, attrCb, conn );
+            }
         }
 
-        for ( const auto& sj : j[ "spaceJoints" ] ) {
-            auto [ destinationModule, destinationComponent ] = details::connectionComponentFromJSON( sj[ "to" ], f );
-
-            Vector fixedPoint = { sj[ "point" ][ 0 ], sj[ "point" ][ 1 ], sj[ "point" ][ 2 ] };
-            if ( sj[ "joint" ][ "type" ] == "rigid" ) {
-                auto conn = connect< RigidJoint >( world.getModule( destinationModule )->components()[ destinationComponent ]
-                                                 , fixedPoint
-                                                 , details::matrixFromJSON( sj[ "joint" ][ "sourceToDestination" ] ) );
-                details::processAttributes( sj, attrCb, conn );
-            } else if ( sj[ "joint" ][ "type" ] == "rotational" ) {
-                auto& jj = sj[ "joint" ];
-                std::vector< float > positions = jj[ "positions" ];
+        if ( j.contains( "spaceJoints" ) ) {
+            for ( const auto& sj : j[ "spaceJoints" ] ) {
+                auto [ destinationModule, destinationComponent ] = details::connectionComponentFromJSON( sj[ "to" ], f );
 
```

Comment:
To stejné - líbilo by se mi pojmenovat. Abych dekomponoval, co to znamená načíst joint a kde ho načítám (resp. odkud).

Reply:
Vyrobí se issue, opraví se v budoucnu.